src/type-hierarchy.ts

```diff
@@ -0,0 +1,213 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as vscodelc from 'vscode-languageclient';
+
+export namespace TypeHierarchyDirection {
+export const Children = 0;
+export const Parents = 1;
+export const Both = 2;
+}
+
+type TypeHierarchyDirection = 0|1|2;
+
+interface TypeHierarchyParams extends vscodelc.TextDocumentPositionParams {
+  resolve?: number;
+  direction: TypeHierarchyDirection;
+}
+
+interface TypeHierarchyItem {
+  name: string;
+  detail?: string;
+  kind: vscodelc.SymbolKind;
+  deprecated?: boolean;
+  uri: string;
+  range: vscodelc.Range;
+  selectionRange: vscodelc.Range;
+  parents?: TypeHierarchyItem[];
+  children?: TypeHierarchyItem[];
+  data?: any;
+}
+
+export namespace TypeHierarchyRequest {
```

Comment:
this doesn't need to be exported? it is just used in this file. the same to `ResolveTypeHierarchyRequest`